analysisservice/src/main/java/nu/yona/server/analysis/service/ActivityCacheService.java

```diff
@@ -16,40 +15,34 @@
 import org.springframework.data.domain.PageRequest;
 import org.springframework.stereotype.Service;
 
+import nu.yona.server.analysis.entities.Activity;
 import nu.yona.server.analysis.entities.DayActivity;
-import nu.yona.server.analysis.entities.WeekActivity;
 
 @Service
 // Day activities are only used in the analysis engine service, so a local cache suffices for as long as we do not
 // scale out the analysis engine service.
-@CacheConfig(cacheManager = "localCache")
+@CacheConfig(cacheManager = "localCache", cacheNames = "lastActivity")
 public class ActivityCacheService
 {
-	@Cacheable(value = "dayActivities", key = "{#userAnonymizedID,#goalID}")
+	@Cacheable(key = "{#userAnonymizedID,#goalID}")
 	@Transactional
-	public DayActivity fetchLastDayActivityForUser(UUID userAnonymizedID, UUID goalID)
+	public ActivityDTO fetchLastActivityForUser(UUID userAnonymizedID, UUID goalID)
 	{
 		List<DayActivity> lastActivityList = DayActivity.getRepository().findLast(userAnonymizedID, goalID, new PageRequest(0, 1))
```

Comment:
This (old) name is wrong. It should be ``lastDayActivities`` or ``lastDayActivityList``.